be/src/runtime/types.cpp

```diff
@@ -261,16 +261,46 @@ std::string TypeDescriptor::debug_string() const {
 }
 
 bool TypeDescriptor::support_join() const {
-    return type != TYPE_JSON && type != TYPE_OBJECT && type != TYPE_PERCENTILE && type != TYPE_HLL &&
-           type != TYPE_MAP && type != TYPE_STRUCT && type != TYPE_ARRAY;
+    if (type == TYPE_ARRAY) {
+        return children[0].support_join();
+    }
+    if (type == TYPE_MAP) {
+        return children[0].support_join() && children[1].support_join();
+    }
+    if (type == TYPE_STRUCT) {
+        for (auto& c : children) {
```

Comment:
suggest to use std::all_of to simplify code

Reply:
changed